starlette/testclient.py

```diff
@@ -514,63 +563,58 @@ def __enter__(self) -> "TestClient":
             def reset_portal() -> None:
                 self.portal = None
 
-            self.stream_send = StapledObjectStream(
-                *anyio.create_memory_object_stream(math.inf)
-            )
-            self.stream_receive = StapledObjectStream(
-                *anyio.create_memory_object_stream(math.inf)
-            )
-            self.task = portal.start_task_soon(self.lifespan)
-            portal.call(self.wait_startup)
+            fut, (task_status, shutdown_event) = portal.start_task(self._lifespan)
 
             @stack.callback
-            def wait_shutdown() -> None:
-                portal.call(self.wait_shutdown)
+            def wait_shutdown() -> bool:
```

Comment:
what should we do here if an exception is raised from this block?

I previously had this cancel the `fut.cancel()` but then that means I can't retrieve any failures from the task:

```python
async def task(*, task_status):
    with anyio.CancelScope() as scope:
        task_status.started(scope)
        try:
            await anyio.sleep_forever()
        except anyio.get_cancelled_exc_type():
            raise MyError

fut, scope = portal.start_task(task)
time.sleep(0.1)
scope.cancel()
assert isinstance(fut.exception(), MyError)
```

Reply:
ah we can avoid the fut interface provided by anyio and cancel the task ourselves
```python
async def task(*, task_status):
    with anyio.CancelScope() as scope:
        task_status.started(scope)
        try:
            await anyio.sleep_forever()
        except anyio.get_cancelled_exc_type():
            raise MyError

fut, scope = portal.start_task(task)
time.sleep(0.1)
portal.call(scope.cancel)
assert isinstance(fut.exception(), MyError)
```